qiskit/transpiler/passes/optimization/optimize_1q_gates.py

```diff
@@ -97,9 +98,6 @@ def run(self, dag):
                     and current_node.op.definition.global_phase
                 ):
                     right_global_phase += current_node.op.definition.global_phase
-                # If there are any sympy objects coming from the gate convert
-                # to numpy.
-                left_parameters = tuple(float(x) for x in left_parameters)
```

Comment:
It would be good to keep the casting to `float` in the cases where it's possible (unless there is a `ParameterExpression` with free parameters). This will help with performance and allow the simplifications around L200 to still apply, even if initially `left_parameters` contained a bound `ParameterExpression` (like `ParameterExpression(2)`). Maybe something like

```python
try:
    left_parameters = tuple(float(x) for x in left_parameters)
except TypeError:
    # If left_parameters contained any unbound Parameters
    pass
```